client/src/dataset/components/MIBParamsForm.tsx

```diff
@@ -42,30 +43,42 @@ const MIBFileParamsForm: React.SFC<MergedProps> = ({
                 <ErrorMessage name="name" />
                 <Field name="name" id="id_name" />
             </Form.Field>
-            <Form.Field>
-                <label htmlFor="id_scan_size_0">Scan Size:</label>
-                <ErrorMessage name="scan_size" />
-                <TupleInput value={values.scan_size} minLen={2} maxLen={2} fieldName="scan_size" setFieldValue={setFieldValue} />
-            </Form.Field>
+            <Reshape navShape={values.nav_shape} sigShape={values.sig_shape} syncOffset={values.sync_offset} imageCount={values.image_count} setFieldValue={setFieldValue} />
             <Button primary={true} type="submit" disabled={isSubmitting || isValidating}>Load Dataset</Button>
             <Button type="button" onClick={onCancel}>Cancel</Button>
             <Button type="button" onClick={handleReset}>Reset</Button>
         </Form>
     )
 }
 
-export default withValidation<DatasetParamsMIB, DatasetParamsMIBForForm>({
+export default withValidation<DatasetParamsMIB, DatasetParamsMIBForForm, DatasetInfoMIB>({
     formToJson: (values, path) => {
         return {
             path,
             type: DatasetTypes.MIB,
             name: values.name,
-            scan_size: parseNumList(values.scan_size),
+            nav_shape: parseNumList(values.nav_shape),
+            sig_shape: parseNumList(values.sig_shape),
+            sync_offset: values.sync_offset,
         }
     },
-    mapPropsToValues: ({path, initial }) => ({
+    mapPropsToValues: ({ path, initial, info }) => ({
         name: getInitialName("name",path,initial),
-        scan_size: getInitial("scan_size", "", initial).toString(),
+        nav_shape: getInitial("nav_shape", "", initial).toString(),
+        sig_shape: getInitial("sig_shape", "", initial).toString(),
+        sync_offset: getInitial("sync_offset", 0, initial),
+        image_count: getInitial("image_count", 0, info),
+        native_sig_shape: getInitial("native_sig_shape", "", info).toString(),
     }),
     type: DatasetTypes.MIB,
+    customValidation: (values) => {
```

Comment:
Validation looks good! Having `native_sig_shape` in the form values is a bit unclean, as it means the values need to be separated again afterwards. Instead, `CustomValidationFn` could also take the `props` as parameter, and would have access to all the `info` fields via `props.info` (`helpers.ts`):


```patch
@@ -65,13 +69,13 @@ export function isKnownDatasetType(detectedType: string) {
 
 type FormToJsonFn<SubmitParams, FormParams> = (inParams: FormParams, path: string) => SubmitParams;
 type PropsToValuesFn<SubmitParams, FormParams, FormInfo> = (props: OpenFormProps<SubmitParams, FormInfo>) => FormParams;
-type CustomValidationFn<FormParams> = (inParams: FormParams) => FormikErrors<FormikValues>;
+type CustomValidationFn<SubmitParams, FormParams, FormInfo> = (inParams: FormParams, props: OpenFormProps<SubmitParams, FormInfo>) => FormikErrors<FormikValues>;
 
 interface WithValidationOpts<SubmitParams extends object, FormParams, FormInfo> {
     formToJson: FormToJsonFn<SubmitParams, FormParams>,
     mapPropsToValues: PropsToValuesFn<SubmitParams, FormParams, FormInfo>,
     type: DatasetTypes,
-    customValidation?: CustomValidationFn<FormParams>
+    customValidation?: CustomValidationFn<SubmitParams, FormParams, FormInfo>
     // WrappedComponent: React.FunctionComponent<FormikProps<FormParams> & OpenFormProps<SubmitParams>>
 }
 
@@ -87,7 +91,7 @@ export function withValidation<SubmitParams extends object, FormParams, FormInfo
             formikBag.setSubmitting(false);
         },
         validate: (values, props) => {
-            return validateOpen(opts.type, opts.formToJson(values, props.path), opts.customValidation?.(values));
+            return validateOpen(opts.type, opts.formToJson(values, props.path), opts.customValidation?.(values, props));
         },
         enableReinitialize: true,
         validateOnChange: true,
```

This is also true for `image_count` btw.: 

```patch
@@ -34,6 +34,7 @@ const MIBFileParamsForm: React.SFC<MergedProps> = ({
     isValidating,
     onCancel,
     setFieldValue,
+    info,
 }) => {
 
     return (
@@ -43,7 +44,7 @@ const MIBFileParamsForm: React.SFC<MergedProps> = ({
                 <ErrorMessage name="name" />
                 <Field name="name" id="id_name" />
             </Form.Field>
-            <Reshape navShape={values.nav_shape} sigShape={values.sig_shape} syncOffset={values.sync_offset} imageCount={values.image_count} setFieldValue={setFieldValue} />
+            <Reshape navShape={values.nav_shape} sigShape={values.sig_shape} syncOffset={values.sync_offset} imageCount={info?.image_count} setFieldValue={setFieldValue} />
             <Button primary={true} type="submit" disabled={isSubmitting || isValidating}>Load Dataset</Button>
             <Button type="button" onClick={onCancel}>Cancel</Button>
             <Button type="button" onClick={handleReset}>Reset</Button>
```

Then, the form values passed to formik are really only those that the user can edit.

Additionally, the code can be improved by extracting the common case (validating sig shape and sync offset) into a helper function, and to use that function instead of copying code. This should be warranted in this case, as there are more than two duplicates.

Reply:
Thanks!